src/components/Class/index.js

```diff
@@ -219,160 +258,246 @@ function Class({ classToEdit }) {
       if (value) {
         if (fieldName === "max_enrolment") {
           formFields[fieldName] = Number(value);
+        } else if (fieldName === "on_days") {
+          formFields[fieldName] = value.split(",");
         } else {
           formFields[fieldName] = value;
         }
       }
     }
-
+    delete formFields.class_type;
     handleTimeValidationAndCreateClass(formFields);
   };
 
   return (
     <div className="ng-create-class">
       <h2 className="title">
-        {isEditMode ? "Update class" : "Create a class"}
+        {isEditMode ? "Update class" : "Create Cohort / Single Class"}
       </h2>
       <Form
         className="form"
         onSubmit={onFormSubmit}
         initialFieldsState={initialFormState}
       >
-        {({ formFieldsState, setFormField }) => {
+        {({ formFieldsState, setFormField, setFormFieldsState }) => {
```

Comment:
why is setFormFieldsState required? We were using `setFormField` to set the state of a field before. Why is that not sufficient?

Reply:
`setFormField` we have to give the key and value one at a time but we need to do `{ ...field, [e.target.name]: e.target.value, until: "" }` to make until empty when it's occurrence and vice versa. Hence, using setFormFieldsState as `setFormFieldsState({ ...field, [e.target.name]: e.target.value, until: "" })`

Reply:
`setFormFieldsState({ ...formFieldsState, [e.target.name]: e.target.value, until: "" })`